arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -674,15 +663,30 @@ void RocksDBHotBackupRestore::execute() {
     if (good) {
       /// Step 2. initiate shutdown and restart with new data directory
       restoreExistingPath = getRocksDBPath();
-      restoreFailsafePath = rebuildPath(dirFailsafeString);
+      if (_saveCurrent) {
+        restoreFailsafePath = buildDirectoryPath(timepointToString(std::chrono::system_clock::now()),
+                                                                   "before_restore");
+        failsafeName = TRI_Basename(restoreFailsafePath.c_str());
+      } else {
+        failsafeName=dirFailsafeString;
+        if (0==failsafeName.compare(_directoryRestore)) {
+          failsafeName += ".1";
+        } // if
+        restoreFailsafePath = rebuildPath(failsafeName);
+      };
```

Comment:
I do not get the logic of this `if .. else ..` block. Maybe add comment which paths are computed andused in both cases.